x-pack/plugins/apm/server/routes/index_pattern.ts

```diff
@@ -13,29 +12,20 @@ import { getDynamicIndexPattern } from '../lib/index_pattern/get_dynamic_index_p
 import { getApmIndices } from '../lib/settings/apm_indices/get_apm_indices';
 
 export const staticIndexPatternRoute = createRoute((core) => ({
-  method: 'POST',
-  path: '/api/apm/index_pattern/static',
+  endpoint: 'POST /api/apm/index_pattern/static',
   handler: async ({ context, request }) => {
     const setup = await setupRequest(context, request);
     const savedObjectsClient = await getInternalSavedObjectsClient(core);
+
     await createStaticIndexPattern(setup, context, savedObjectsClient);
 
     // send empty response regardless of outcome
     return undefined;
   },
 }));
 
-export const dynamicIndexPatternRoute = createRoute(() => ({
-  path: '/api/apm/index_pattern/dynamic',
-  params: {
-    query: t.partial({
-      processorEvent: t.union([
-        t.literal('transaction'),
-        t.literal('metric'),
-        t.literal('error'),
-      ]),
-    }),
-  },
+export const dynamicIndexPatternRoute = createRoute({
```

Comment:
Shouldn't we be passing `processorEvent ` along to `getDynamicIndexPattern`? If not will you remove `processorEvent` from `getDynamicIndexPattern` and `getPatternIndices`?

Reply:
Probably also remove it from here https://github.com/elastic/kibana/blob/00c053594682a0f66160150e61df91296aa87312/x-pack/plugins/apm/server/lib/helpers/setup_request.ts#L63

Reply:
I've put it back, we were not using it but looks like it was a bug I introduced in https://github.com/elastic/kibana/commit/429805d1b878919c0b7be081f425d5dc0ec67c09. Previously, we were using `query.processorEvent` in `setup_request`, but we moved it out of `setup_request` into its own function, which required the query param to be passed explicitly, but looked like that never happened.